pkg/servicemesh/federation/federation.go

```diff
@@ -72,6 +75,11 @@ func New(opt Options) (*Federation, error) {
 	if err := opt.validate(); err != nil {
 		return nil, err
 	}
+	name := types.NamespacedName{Name: os.Getenv("POD_NAME"), Namespace: os.Getenv("POD_NAMESPACE")}
```

Comment:
Do you have access to pilotArgs here? If yes, this struct already have these variables for free: https://github.com/maistra/istio/blob/8693f7adc22cdb8ee2db0854519320bece8bad84/pilot/pkg/bootstrap/options.go#L152

Reply:
I thought I investigated this, but I'll check again.

Reply:
updated